interactive_engine/executor/store/src/api/property.rs

```diff
@@ -44,6 +44,51 @@ pub enum Property {
     Unknown,
 }
 
+impl PartialOrd for Property {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        match (self, other) {
+            (Property::Bool(left), Property::Bool(right)) => left.partial_cmp(right),
+            (Property::Char(left), Property::Char(right)) => left.partial_cmp(right),
+            // cmp between integer and float
+            // if both are integers, cast to long
+            // else cast to double
+            (Property::Short(_), _)
+            | (Property::Int(_), _)
+            | (Property::Long(_), _)
+            | (_, Property::Short(_))
+            | (_, Property::Int(_))
+            | (_, Property::Long(_)) => {
+                if self.is_float_type() || other.is_float_type() {
+                    let left = self.get_double().expect("Convert error");
+                    let right = other.get_double().expect("Convert error");
+                    left.partial_cmp(&right)
+                } else {
+                    let left = self.get_long().expect("Convert error");
+                    let right = other.get_long().expect("Convert error");
+                    left.partial_cmp(&right)
+                }
+            }
+            (Property::Float(left), Property::Float(right)) => left.partial_cmp(right),
+            (Property::Double(_), _) | (_, Property::Double(_)) => {
+                let left = self.get_double().expect("Convert error");
+                let right = other.get_double().expect("Convert error");
+                left.partial_cmp(&right)
+            }
+            (Property::Bytes(left), Property::Bytes(right)) => left.partial_cmp(right),
+            (Property::String(left), Property::String(right)) => left.partial_cmp(right),
+            (Property::Date(left), Property::Date(right)) => left.partial_cmp(right),
+            (Property::ListInt(left), Property::ListInt(right)) => left.partial_cmp(right),
```

Comment:
As above, we compare integers/floats by converting into `long`/`double`. How about compare between  e.g., `Property::ListInt` and `Property::ListLong`?

Reply:
now cast number lists to cmp